File: trajectory_generator/scripts/trajectory_generator_gripper_client.py
#!/usr/bin/env python

import sys
import rospy
from trajectory_generator.srv import *

def trajectory_generator_gripper_client(x, y):
    rospy.wait_for_service('trajectory_generator_gripper_server')
    try:
        trajectory_generator_gripper = rospy.ServiceProxy('trajectory_generator_gripper', trajectory_generator_gripper)
        resp1 = trajectory_generator_gripper(x, y)
        return resp1.sum
    except rospy.ServiceException, e:
        print "Service call failed: %s"%e

def usage():
    return "%s [x y]"%sys.argv[0]

if __name__ == "__main__":
    if len(sys.argv) == 3:
        x = int(sys.argv[1])
        y = int(sys.argv[2])
    else:
        print usage()
        sys.exit(1)
    print "Requesting %s+%s"%(x, y)
    print "%s + %s = %s"%(x, y, trajectory_generator_gripper_client(x, y))

File: trajectory_generator/src/trajectory_generator_gripper_client.cpp
#include "ros/ros.h"
#include "trajectory_generator/trajectory_generator_gripper.h"
#include <cstdlib>

int main(int argc, char **argv)
{
  ros::init(argc, argv, "trajectory_generator_client");
  if (argc != 3)
  {
    ROS_INFO("usage: trajectory_generator_client X Y");
    return 1;
  }

  ros::NodeHandle n;
  ros::ServiceClient client = n.serviceClient<trajectory_generator::trajectory_generator_gripper>("trajectory_generator_gripper");
  trajectory_generator::trajectory_generator_gripper srv;
  srv.request.a = atoll(argv[1]);
  srv.request.b = atoll(argv[2]);
  if (client.call(srv))
  {
    ROS_INFO("Sum: %ld", (long int)srv.response.sum);
  }
  else
  {
    ROS_ERROR("Failed to call service trajectory_generator_gripper");
    return 1;
  }

  return 0;
}